### ISSUE TITLE
Build failure on armv7l-linux and riscv64-linux (11.2.0, 12.0.0) (`error: 'batch_bool_cast' was not declared in this scope`)

### ISSUE THREAD

**Issue by misuzu** [xtensor-stack/xsimd#986]
xsimd fails to build on armv7l-linux and riscv64-linux since [11.2.0](https://github.com/xtensor-stack/xsimd/releases/tag/11.2.0), x86_64-linux and aarch64-linux have no issues

```
<.................>
[  5%] Building CXX object test/CMakeFiles/test_xsimd.dir/test_api.cpp.o
In file included from /build/source/include/xsimd/types/../arch/./xsimd_generic.hpp:18,
                 from /build/source/include/xsimd/types/../arch/xsimd_isa.hpp:92,
                 from /build/source/include/xsimd/types/xsimd_batch.hpp:492,
                 from /build/source/include/xsimd/xsimd.hpp:61,
                 from /build/source/test/test_api.cpp:12:
/build/source/include/xsimd/types/../arch/././generic/xsimd_generic_math.hpp: In function 'xsimd::batch<float, A> xsimd::kernel::cbrt(const xsimd::batch<float, A>&, requires_arch<xsimd::generic>)':
/build/source/include/xsimd/types/../arch/././generic/xsimd_generic_math.hpp:152:45: error: 'batch_bool_cast' was not declared in this scope; did you mean 'batch_cast'?
  152 |             const batch_type cbrt2 = select(batch_bool_cast<float>(flag), CBRT2, CBRT2I);
      |                                             ^~~~~~~~~~~~~~~
      |                                             batch_cast
/build/source/include/xsimd/types/../arch/././generic/xsimd_generic_math.hpp:152:61: error: expected primary-expression before 'float'
  152 |             const batch_type cbrt2 = select(batch_bool_cast<float>(flag), CBRT2, CBRT2I);
      |                                                             ^~~~~
/build/source/include/xsimd/types/../arch/././generic/xsimd_generic_math.hpp:153:61: error: expected primary-expression before 'float'
  153 |             const batch_type cbrt4 = select(batch_bool_cast<float>(flag), CBRT4, CBRT4I);
      |                                                             ^~~~~
/build/source/include/xsimd/types/../arch/././generic/xsimd_generic_math.hpp:154:54: error: expected primary-expression before 'float'
  154 |             batch_type fact = select(batch_bool_cast<float>(rem == i_type(1)), cbrt2, batch_type(1.));
      |                                                      ^~~~~
/build/source/include/xsimd/types/../arch/././generic/xsimd_generic_math.hpp:155:43: error: expected primary-expression before 'float'
  155 |             fact = select(batch_bool_cast<float>(rem == i_type(2)), cbrt4, fact);
<.................>
```

Build logs:
[xsimd-12.0.0-armv7l-linux.log](https://github.com/xtensor-stack/xsimd/files/13549548/xsimd-12.0.0-armv7l-linux.log)
[xsimd-12.0.0-riscv64-linux.log](https://github.com/xtensor-stack/xsimd/files/13549549/xsimd-12.0.0-riscv64-linux.log)

Maybe related? https://github.com/xtensor-stack/xsimd/issues/716

**Comment by serge-sans-paille** [comment]
Can you confirm this got fixed by #991 ?

**Comment by misuzu** [comment]
> Can you confirm this got fixed by #991 ?

It did, thank you!
